pkg/config/security/security_test.go

```diff
@@ -116,6 +116,15 @@ func TestValidateCondition(t *testing.T) {
 			values:    []string{"a.b.c.d"},
 			wantError: true,
 		},
+		{
```

Comment:
Not related to this part but just commented here, could you try to add some e2e tests (`tests/integration/security/authorization_test.go`) for the new `remote.ip` field?  

I think you first need to configure the `numTrustedProxies` (https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/#configuring-x-forwarded-for-headers) and then can probably add the `X-Forwarded-For` header manually the test so simulate the remote IP.

Reply:
I fixed the unit tests and added some integration tests: TestAuthorization_IngressGatewayRemoteIp, TestAuthorization_IngressGatewayNotRemoteIp, and TestAuthorization_IngressGatewayRemoteIpAttr.  

I have not been able to get integration tests to successfully run in my dev environment.  All of the requests to the ingress gateway from the framework timeout.  Is it easy for you to run them and see if they succeed?

Reply:
Ok, I finally got tests to run, and with some tweaks the new tests I mentioned do seem to work.